File: apple/tvos/tvOSSampleApp_ObjC/tvOSSampleApp_ObjC/CollectionInfoViewController.m
//
//  CollectionInfoViewController.m
//  tvOSSampleApp_ObjC
//
//  Created by Patrick McConnell on 2/18/16.
//  Copyright © 2016 Patrick McConnell. All rights reserved.
//

#import "CollectionInfoViewController.h"
#import "CollectionInfoVideoTableViewController.h"


@interface CollectionInfoViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *thumbnailImageView;
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet UITextView *notesTextView;
@end

@implementation CollectionInfoViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  [self configureForCollection];
}

//func configureForCollection() {
- (void) configureForCollection {
  dispatch_async(dispatch_get_main_queue(), ^{
    [self.titleLabel setText:self.collection.title];
    [self.notesTextView setText: self.collection.notes];
    
    [self.collection thumbnail:^(UIImage* image) {
      if (image != nil) {
        [self.thumbnailImageView setImage:image];
      }
    }];
  });
}


-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
  if ([segue.identifier isEqualToString:@"videoTableEmbed"]) {
    CollectionInfoVideoTableViewController* vc = segue.destinationViewController;
    
    [[OddContentStore sharedStore]objectsOfType:OddMediaObjectTypeVideo ids:self.collection.objectIds callback:^(NSArray* objects) {
      if ( objects  != nil ) {
        vc.videos = objects;
      }
    }];
    
  }
}
@end

File: apple/tvos/tvOSSampleApp/tvOSSampleApp/CollectionInfoViewController.swift
//
//  CollectionInfoViewController.swift
//  tvOSSampleApp
//
//  Created by Patrick McConnell on 1/28/16.
//  Copyright © 2016 Odd Networks. All rights reserved.
//

import UIKit
import OddSDKtvOS

class CollectionInfoViewController: UIViewController {
  
  @IBOutlet var collectionThumbnailImageView: UIImageView!
  @IBOutlet var collectionTitleLabel: UILabel!
  @IBOutlet weak var collectionNotesTextView: UITextView!
  
  var collection = OddMediaObjectCollection() {
    didSet {
      configureForCollection()
    }
  }
  
  override func viewDidLoad() {
    super.viewDidLoad()
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }
  
  

  // MARK: - Navigation
  
  override func prepare(for segue: UIStoryboardSegue, sender: AnyObject?) {
    guard let id = segue.identifier else { return }
    switch id {
    case "videoTableEmbed":
      guard let vc = segue.destination as? CollectionInfoVideoTableViewController,
      let node = self.collection.relationshipNodeWithName("entities"),
        let ids = node.allIds else { break }
      
      OddContentStore.sharedStore.objectsOfType(.video, ids: ids, include:nil, callback: { (objects, errors) -> Void in
        if let videos = objects as? Array<OddVideo> {
          vc.videos = videos
        }
      })
      
    default:
      break
    }

  }


  // MARK: - Helpers
  
  func configureForCollection() {
    DispatchQueue.main.async { () -> Void in
      self.collectionTitleLabel?.text = self.collection.title
      self.collectionNotesTextView?.text = self.collection.notes
      
      self.collection.thumbnail { (image) -> Void in
        if let thumbnail = image {
          self.collectionThumbnailImageView?.image = thumbnail
        }
      }
    }
    
  }
  
}
